dev-java/protobuf-java/protobuf-java-4.28.0.ebuild

```
@@ -0,0 +1,161 @@
+# Copyright 2008-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+JAVA_PKG_IUSE="doc source test"
+MAVEN_ID="com.google.protobuf:protobuf-java:${PV}"
+JAVA_TESTING_FRAMEWORKS="junit-4"
+
+inherit java-pkg-2 java-pkg-simple cmake
+
+DESCRIPTION="Core Protocol Buffers library"
+HOMEPAGE="https://protobuf.dev"
+# Currently we bundle the binary version of truth.jar used only for tests, we don't install it.
+# And we build artifact 4.28.0 from the 28.0 tarball in order to allow sharing the tarball with
+# dev-libs/protobuf.
+MY_PV4="${PV#4.}"
+MY_PV="${MY_PV4/_rc/-rc}"
+SRC_URI="https://github.com/protocolbuffers/protobuf/archive/v${MY_PV}.tar.gz -> protobuf-${MY_PV}.tar.gz
```

```suggestion
SRC_URI="https://github.com/protocolbuffers/protobuf/releases/download/v${PV}/${P}.tar.gz"
```

The old one is a snapshot of the tag. That can change and might lead to trouble. I'm changing the rest of protobuf as well.

---

saving the download as `4.28.0.tar.gz` instead of `28.0.tar.gz` would most probably cause a `MatchingChksums` complaint in ci (in case the corresponding `dev-libs/protobuf` got added)

---

```suggestion
https://github.com/protocolbuffers/protobuf/releases/download/v${PROTOBUF_PV}/${PN%%-java}-${PROTOBUF_PV}.tar.gz
```
I copied from the wrong file. Same reason though.